### PR TITLE
Fix Copy linked files dialog showing NPE when no linked files

### LINKED ISSUE #3818: Exception when trying to save linked file
- [X ] I have tested the latest master version from http://builds.jabref.org/master/ and the problem persists

JabRef 4.2-dev--snapshot--2018-03-06--master--bb471a26d
Linux 4.4.0-104-generic amd64 
Java 1.8.0_161

Steps to reproduce:

1. Create an entry with a URL leading to a PDF file, like this: https://www.gla.ac.uk/media/media_452489_en.pdf
2. Right-click the entry in the entry table.
3. Choose "Copy linked files to folder...".
4. Choose your home folder.

<details>
File: copyfilesdialog.fxml not found, attempting with camel case
10:16:05.694 [JavaFX Application Thread] ERROR org.jabref.FallbackExceptionHandler - Uncaught exception occurred in Thread[JavaFX Application Thread,5,main]
java.lang.IllegalStateException: Cannot load org.jabref.gui.copyfiles.copyfilesdialog
	at com.airhacks.afterburner.views.FXMLView.loadSynchronously(FXMLView.java:101) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at com.airhacks.afterburner.views.FXMLView.initializeFXMLLoader(FXMLView.java:125) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at com.airhacks.afterburner.views.FXMLView.getView(FXMLView.java:137) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at org.jabref.gui.AbstractView.getView(AbstractView.java:28) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at org.jabref.gui.copyfiles.CopyFilesDialogView.show(CopyFilesDialogView.java:25) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at org.jabref.gui.copyfiles.CopyFilesAction.showDialog(CopyFilesAction.java:63) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at org.jabref.gui.copyfiles.CopyFilesAction.lambda$startServiceAndshowProgessDialog$3(CopyFilesAction.java:58) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at com.sun.javafx.application.PlatformImpl.lambda$null$172(PlatformImpl.java:295) ~[jfxrt.jar:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_161]
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$173(PlatformImpl.java:294) ~[jfxrt.jar:?]
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95) ~[jfxrt.jar:?]
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method) ~[jfxrt.jar:?]
	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$48(GtkApplication.java:139) ~[jfxrt.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_161]
Caused by: javafx.fxml.LoadException: 
file:/home/erelsgl/Dropbox/papers/JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar!/org/jabref/gui/copyfiles/CopyFilesDialog.fxml

	at javafx.fxml.FXMLLoader.constructLoadException(FXMLLoader.java:2601) ~[jfxrt.jar:?]
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2571) ~[jfxrt.jar:?]
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2441) ~[jfxrt.jar:?]
	at javafx.fxml.FXMLLoader.load(FXMLLoader.java:2409) ~[jfxrt.jar:?]
	at com.airhacks.afterburner.views.FXMLView.loadSynchronously(FXMLView.java:99) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	... 13 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_161]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_161]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_161]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_161]
	at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71) ~[?:1.8.0_161]
	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_161]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_161]
	at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275) ~[?:1.8.0_161]
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2566) ~[jfxrt.jar:?]
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2441) ~[jfxrt.jar:?]
	at javafx.fxml.FXMLLoader.load(FXMLLoader.java:2409) ~[jfxrt.jar:?]
	at com.airhacks.afterburner.views.FXMLView.loadSynchronously(FXMLView.java:99) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	... 13 more
Caused by: java.lang.NullPointerException
	at java.util.AbstractCollection.addAll(AbstractCollection.java:343) ~[?:1.8.0_161]
	at javafx.collections.ModifiableObservableListBase.addAll(ModifiableObservableListBase.java:99) ~[jfxrt.jar:?]
	at javafx.beans.binding.ListExpression.addAll(ListExpression.java:266) ~[jfxrt.jar:?]
	at org.jabref.gui.copyfiles.CopyFilesDialogViewModel.<init>(CopyFilesDialogViewModel.java:14) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at org.jabref.gui.copyfiles.CopyFilesDialogController.initialize(CopyFilesDialogController.java:34) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_161]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_161]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_161]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_161]
	at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71) ~[?:1.8.0_161]
	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_161]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_161]
	at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275) ~[?:1.8.0_161]
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2566) ~[jfxrt.jar:?]
	at javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2441) ~[jfxrt.jar:?]
	at javafx.fxml.FXMLLoader.load(FXMLLoader.java:2409) ~[jfxrt.jar:?]
	at com.airhacks.afterburner.views.FXMLView.loadSynchronously(FXMLView.java:99) ~[JabRef-4.2-dev--snapshot--2018-03-06--master--bb471a26d.jar:?]
	... 13 more

</details>

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes #3818 
(ignore the branch name, I  thought it was a different error)

The part in the CopyTask should also be copied to the maintable beta branch. The rest is already there.

----

- [x] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/copyfiles/CopyFilesTask.java [hunk 1]
```diff
@@ -103,6 +103,10 @@ public CopyFilesTask(BibDatabaseContext databaseContext, List<BibEntry> entries,
                 }
             }
             updateMessage(Localization.lang("Finished copying"));
+
+            if (newPath == null) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/copyfiles/CopyFilesTask.java]: `newPath` is an optional, so it should never be null (for the same reason as for collections). Please init `newPath` as `empty` (and then this if test should not be necessary).

File: src/main/java/org/jabref/gui/copyfiles/CopyFilesAction.java [hunk 2]
```diff
@@ -52,13 +52,20 @@ public void actionPerformed(ActionEvent e) {
...
+        });
+
+        exportService.run();
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+            showDialog(exportService.getValue());
+        });
     }
 
     private void showDialog(List<CopyFilesResultItemViewModel> data) {
+
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/copyfiles/CopyFilesAction.java]: What's the reason for the newlines?

File: src/main/java/org/jabref/gui/copyfiles/CopyFilesTask.java (hunk 1 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/gui/copyfiles/CopyFilesTask.java]: Ah thanks! didn't catch that it was an optional

**Comment by Siedlerchr** [comment]
travis fails for some timeout reason not related

**Pull request merged**